Disallow calls to vtkOpenGLRenderWindow::New

By default this function probes all rendering backend one by one,
spewing warnings as it goes for anything that doesn't work, and
uses a process-wide environment variable to govern a preference.

Both of those disqualify it from use in Drake, so we lobomotize it to
prevent it from ever being accidentally used.

This patch should not be upstreamed.

--- Rendering/OpenGL2/vtkOpenGLRenderWindow.cxx
+++ Rendering/OpenGL2/vtkOpenGLRenderWindow.cxx
@@ -501,7 +501,12 @@ vtkOpenGLRenderWindow::~vtkOpenGLRenderWindow()
   this->State->Delete();
 }
 
-#if !(defined(__APPLE__) || defined(__ANDROID__) || defined(__EMSCRIPTEN__))
+#if 1  // This function is too unreliable for Drake's purposes, so we'll nix it.
+vtkOpenGLRenderWindow* vtkOpenGLRenderWindow::New()
+{
+  throw std::runtime_error("Internal error: Drake must not use vtkOpenGLRenderWindow");
+}
+#else
 //------------------------------------------------------------------------------
 vtkOpenGLRenderWindow* vtkOpenGLRenderWindow::New()
 {
